### ISSUE TITLE
TextSelector overlay isn't hidden when selected text exits viewport

### ISSUE THREAD

**Issue by huangyichieh** [flutter/flutter#89570]
## Steps to Reproduce

minimal reproducible example:
```dart
void main() {
  runApp(MyHomePage());
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter test',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        primaryColor: Colors.green[900],
      ),
      home: Scaffold(
        appBar: AppBar(
          title: Text("Flutter test"),
        ),
        body: Container(
          color: Colors.grey,
          constraints: BoxConstraints(
            maxHeight: 200
          ),
          child: SingleChildScrollView(
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: Column(
                children: [
                  SelectableText(
                    "test1\ntest2\ntest3\ntest4\ntest5\ntest6\ntest7\ntest8\ntest9\ntest10\ntest11\ntest12\ntest13\ntest14\ntest15\ntest16",
                  ),
                  Divider(),
                  SelectableText(
                    "test1\ntest2\ntest3\ntest4\ntest5\ntest6\ntest7\ntest8\ntest9\ntest10\ntest11\ntest12\ntest13\ntest14\ntest15\ntest16",
                  ),
                  Divider(),
                  SelectableText(
                    "test1\ntest2\ntest3\ntest4\ntest5\ntest6\ntest7\ntest8\ntest9\ntest10\ntest11\ntest12\ntest13\ntest14\ntest15\ntest16",
                  )
                ]
              ),
            )
          ),
        )
      )
    );
  }
}
```
1. build example below.
2. Select any text.
3. Scroll up or down until edge of container.
4. ![Screenshot from 2021-09-07 12-15-29](https://user-images.githubusercontent.com/60538542/132284318-8d078172-2d12-46eb-bc01-4b5f1b98e89e.png)

**Expected results:** The text selection should be hidden when it is out of container

**Actual results:**  The text selection is shown even it is out of container, and stacked in front of other widgets

<details>
  <summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel stable, 2.2.3, on Linux, locale en_US.UTF-8)
    • Flutter version 2.2.3 at /home/jay/Documents/flutter_linux_2.0.3-stable/flutter
    • Framework revision f4abaa0735 (10 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /home/jay/Android/Sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /usr/local/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /usr/local/android-studio
    • Flutter plugin version 55.0.1
    • Dart plugin version 201.9335
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.55.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.22.0

[✓] Connected device (2 available)
    • sdk gphone x86 arm (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 91.0.4472.114
```
</details>

**Comment by danagbemava-nc** [comment]
Issue is reproducible on `master` and `stable`. 

See below for comparisons between `native` and `flutter`

<details>

<summary>native and flutter reproduction</summary>


https://user-images.githubusercontent.com/88313112/132323730-7f8352f4-ff86-4b67-b255-fe7b8d7940cc.MP4 

https://user-images.githubusercontent.com/88313112/132323356-2b694e78-b8fa-4b3f-9e18-7fcd531809e7.MP4

https://user-images.githubusercontent.com/88313112/132323219-d9a6d332-0b93-465f-b510-9c490f04afae.mp4  

https://user-images.githubusercontent.com/88313112/132323267-04810647-0208-44e3-a3f1-10b7e485a125.mp4 


</details>


<details>
<summary>flutter doctor -v</summary>

```console
[✓] Flutter (Channel master, 2.6.0-1.0.pre.193, on macOS 11.5.1 20G80
    darwin-arm, locale en-GB)
    • Flutter version 2.6.0-1.0.pre.193 at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 0c5431d99c (31 hours ago), 2021-09-05 22:31:02 -0400
    • Engine revision b9c633900e
    • Dart version 2.15.0 (build 2.15.0-82.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.5.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E • ios            • iOS
      14.7.1 18G82
    • macOS (desktop)        • macos                     • darwin-arm64   •
      macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 93.0.4577.63

• No issues found!
```

```console
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.1 20G80 darwin-arm, locale
    en-GB)
    • Flutter version 2.2.3 at /Users/nexus/dev/sdks/flutter
    • Framework revision f4abaa0735 (10 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (3 available)
    • Nexus’ IPhone (mobile) • 00008020-001875E83A38002E • ios            • iOS
      14.7.1
    • macOS (desktop)        • macos                     • darwin-arm64   •
      macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 93.0.4577.63

• No issues found!
```
</details>

**Comment by huangyichieh** [comment]
Looks like similar issues had happened and had been fixed in earlier version?
#77889
#55547

**Comment by HansMuller** [comment]
CC @chunhtai

**Comment by Camilocubillosd** [comment]
Any update about this issue?  I have the same problem

**Comment by andromeda-center** [comment]
Any update?

**Comment by huycozy** [comment]
I think this issue can be solved once https://github.com/flutter/flutter/issues/52425 landed (_Text selection menu should fade during scroll_).

**Comment by gnprice** [comment]
It looks like this is the same issue as #13182, so let's consolidate discussion on that thread.

**Comment by gnprice** [comment]
> I think this issue can be solved once #52425 landed (_Text selection menu should fade during scroll_).

From looking at the "Native Android" video on that issue (showing desired behavior), even once that issue is fixed it looks like we'd still expect the selection handles at each end of the selection to remain visible when scrolling. So this issue would get a bit milder — the text selection menu with "Copy" and so on would disappear on scroll and so wouldn't typically have this symptom — but would still be present because the selection handles would be unaffected.
